htlcswitch/link.go

```diff
@@ -180,6 +180,11 @@ type ChannelLinkConfig struct {
 	// clients have been restarted, or remote peer have been reconnected.
 	SyncStates bool
 
+	// SkipNonceInit is used during channel reestablish to prevent
+	// ProcessChanSyncMsg from calling InitRemoteMusigNonces if we have
+	// already called it before in NewLightningChannel.
+	SkipNonceInit bool
```

Comment:
Instead of a new config value here, what if we just expose similar data, but via a new method on the channel state machine? That'll give us one less thing to set, and also simplify the control flow for the caller (this value doesn't need to be set).

Reply:
Think I've done this, now uses an exposed function `SkipNonceInit` and an atomic.Bool. Diff is a lot slimmer